### DISCUSSION TITLE
Allowing sudo command as notification action

### DISCUSSION THREAD

**Discussion by ghost** [phuhl/linux_notification_center#250] [Q&A] [answered]
This may not be so much related to Deadd as I'm new to LInux, but I cannot find a consensus on how to execute notification actions that require sudo, for example a button that run "sudo shutdown" in the notification center.

I've tried running the deadd-notification-center as sudo without much success but I don't even know if that's the proper way to do that.
```
(base) alexandre:~/.config/qtile/scripts$ deadd-notification-center
No existing translations for fr_FR.UTF-8.
Consider contributing your language.
https://github.com/phuhl/linux_notification_center/tree/master/translation
Trying English instead...

notificationDaemon started 
^C^C
(base) alexandre:~/.config/qtile/scripts$ sudo deadd-notification-center
No existing translations for fr_FR.UTF-8.
Consider contributing your language.
https://github.com/phuhl/linux_notification_center/tree/master/translation
Trying English instead...

notificationDaemon started
deadd-notification-center: ClientError {clientErrorMessage = "connectSession: DBUS_SESSION_BUS_ADDRESS is invalid.", clientErrorFatal = True}
^C^C
(base) alexandre:~/.config/qtile/scripts$ 
(base) alexandre:~/.config/qtile/scripts$ sudo -E deadd-notification-center
No existing translations for fr_FR.UTF-8.
Consider contributing your language.
https://github.com/phuhl/linux_notification_center/tree/master/translation
Trying English instead...

notificationDaemon started
deadd-notification-center: ClientError {clientErrorMessage = "Error reading message from socket: UnmarshalError \"Unexpected end of input while parsing message header.\"", clientErrorFatal = True}
deadd-notification-center: ClientError {clientErrorMessage = "Network.Socket.sendBuf: invalid argument (Mauvais descripteur de 
(base) alexandre:~/.config/qtile/scripts$
```
As you can see here I can launch the deadd-notification-center without issue but as soon as I use sudo it doesn't work and I don't see simple solutions (but again I'm not experimented in Linux so maybe that's just that).

Any help would be appreciated, thanks a lot :pray:

**Comment by phuhl** [comment, marked as answer]
Hey, you'll want to look into the sudoers file to allow running shutdown without the use of sudo or without the use of a password. Take a look at this Stackoverflow post: https://askubuntu.com/questions/168879/shutdown-from-terminal-without-entering-password/168885#168885
